src/tests/adr/arrays_adr.rs

```diff
@@ -0,0 +1,131 @@
+use crate::test_utils::tests::codegen;
+
+/// # Architecture Design Record: Arrays
```

Comment:
Do we need an entry on the different start and end indices of an array? We usually convert an array access to a 0 based one during code gen

Reply:
good idea

Reply:
done